api/src/org/apache/cloudstack/api/command/admin/usage/GetUsageRecordsCmd.java

```diff
@@ -111,6 +111,30 @@ public Long getProjectId() {
     public String getUsageId() {
         return usageId;
     }
+    public void setAccountName(String accountName) {
+        this.accountName = accountName;
+    }
+
+    public void setDomainId(Long domainId) {
+        this.domainId = domainId;
+    }
+
+    public void setEndDate(Date endDate) {
+        this.endDate = endDate;
+    }
+
+    public void setStartDate(Date startDate) {
+        this.startDate = startDate;
```

Comment:
@abhinandanprateek it's not a matter of it being modified in that time period.  Since the value being passed is simply a pointer, any modification to that value after calling this method will change the value.  All dates should be defensively copied for this reason.